server/mediaserver.go

```diff
@@ -410,9 +410,9 @@ func (s *LivepeerServer) registerConnection(rtmpStrm stream.RTMPVideoStream) (*r
 		Bitrate:    "4000k", // Fix this
 	}
 	hlsStrmID := core.MakeStreamID(mid, &vProfile)
-	s.connectionLock.Lock()
+	s.connectionLock.RLock()
```

Comment:
Maybe add a comment explaining that we do an early check for session existence here because it may be a while between initializing a session and assigning the session to the connection map.

Reply:
Added comments in db802aeba273c686aaa8e0f58d03c65983830fe6